src/backend/pipeline/cont_combiner.c

```diff
@@ -862,6 +863,16 @@ ContinuousQueryCombinerMain(void)
 		if (MyContQueryProc->group->terminate)
 			break;
 
+		/* If necessary, reallocate resources that depend on runtime parameters that were changed */
+		run_params = GetContQueryRunParams();
+		if (run_params->batch_size != group_hashes_len)
+		{
+			MemoryContextSwitchTo(run_cxt);
+			group_hashes_len = run_params->batch_size;
+			group_hashes = repalloc(group_hashes, group_hashes_len * sizeof(int64));
```

Comment:
Isn't `group_hashes` going to be `NULL` for the first run? That would cause `repalloc` to fail.

```
group_hashes_len = run_params->batch_size;
if (group_hashes == NULL)
{
  MemoryContext old = MemoryContextSwitchTo(run_cxt);
  group_hashes = palloc0(group_hashes_len * sizeof(int64));
}
else
{
  group_hashes = repalloc(group_hashes, group_hashes_len * sizeof(int64));
  MemSet(group_hashes, 0, group_hashes_len * sizeof(int64));
}
```